src/dataprocessor/dataprocessor.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Config from '../config.js';
+
+/**
+ * Abstract base DataProcessor class. This class should be extended by actual DataProcessor implementations.
+ *
+ * @abstract
+ * @class dataProcessor.DataProcessor
+ */
+export default class DataProcessor {
```

Comment:
We can consider turning this into an interface. For now, DP contains no code at all.